components/selection/selection-mixin.js

```diff
@@ -96,15 +103,16 @@ export const SelectionMixin = superclass => class extends RtlMixin(CollectionMix
 			this._selectionSelectables.forEach(selectable => {
 				if (selectable.selected) keys.push(selectable.key);
 				if (selectable._indeterminate) state = SelectionInfo.states.some;
+				if (selectable.disabled && !selectable.selected) numDisabledKeys++;
 			});
 
 			if (keys.length > 0) {
-				if (keys.length === this._selectionSelectables.size) state = SelectionInfo.states.all;
+				if ((keys.length + numDisabledKeys) === this._selectionSelectables.size) state = SelectionInfo.states.all;
```

Comment:
Hmmm, I'm not sure about this. Previously if there was a disabled unselected item, the `SelectionInfo` `state` would have been `some`. With this change, it looks like it would be `all`, which isn't really true, and consumers need to know how to deal with that... they can't just assume that `all` means all, where previously they could. I suppose it's consistent with the select all-pages logic immediately above, where it completely ignores disabled unselected items and just reports `allPages`.

There's probably not a lot of usages of disabled list-items, but I'm still not sure if this the state that should be returned to the user, and it seems like it could be a breaking change. 🤔

Reply:
Yes - this does change behaviour, I was uncertain what we'd wanted to guide here as this does bring the pattern to match other places but might be breaking like you said. 

Should we discuss with design on this?
@margaree tagging you since DB is off this week I believe

Reply:
Yup, we should bring Jeff into this discussion. It does seem like this could cause some problems and just as a user I find the "all" checked state a bit misleading when there's a disabled non-selected item.

Reply:
To update this conversation, I've tapped Jeff and we'll discuss this further - I'll update this thread with the results.

Reply:
Jeff remarks:
- Keep old behaviour
- He'd be okay with one day discussing further changing to the behaviour modelled currently
  - but it's not what the scope of this is meant to address so just move along